src/parser.ts

```diff
@@ -466,4 +476,10 @@ export function parseFromProgram(
 		const comment = ts.displayPartsToString(symbol.getDocumentationComment(checker));
 		return comment ? comment : undefined;
 	}
+
+	function getSymbolFileNames(symbol: ts.Symbol): Set<string> {
+		const declarations = symbol.getDeclarations() || [];
```

Comment:
Would it be enough to only get the fileName of the first declaration?
The parser does that here https://github.com/merceyz/typescript-to-proptypes/blob/184621387a67fc98f2c173ee4afc3b149555f517/src/parser.ts#L282

Reply:
I don't know if the order of the declarations is deterministic. I rather put every in their for better customization. In the end I want to intersect function signatures for a complete props type anyway.

What is the purpose of only having the fileName of a single declaration?

Reply:
> What is the purpose of only having the fileName of a single declaration?

We only get the type for one of them, and in this case the prop is probably only in one location. I'm fine with it either way, just curious

Reply:
It may be enough to use a single declaration. I just don't know how the declarations are ordered and if this order is part of the public API.